python/hsml/engine/model_engine.py

```diff
@@ -88,40 +88,73 @@ def _upload_additional_resources(self, model_instance):
             model_instance.model_schema = None
         return model_instance
 
-    def _copy_hopsfs_model(self, existing_model_path, model_version_path):
+    def _copy_or_move_hopsfs_model(
+        self,
+        existing_model_path,
+        model_version_path,
+        keep_original_files,
+        update_upload_progress,
+    ):
         # Strip hdfs prefix
         if existing_model_path.startswith("hdfs:/"):
             projects_index = existing_model_path.find("/Projects", 0)
             existing_model_path = existing_model_path[projects_index:]
 
+        n_dirs, n_files = 0, 0
         for entry in self._dataset_api.list(existing_model_path, sort_by="NAME:desc")[
             "items"
         ]:
             path = entry["attributes"]["path"]
             _, file_name = os.path.split(path)
-            self._dataset_api.copy(path, model_version_path + "/" + file_name)
+            if keep_original_files:
+                self._dataset_api.copy(path, model_version_path + "/" + file_name)
```

Comment:
may be I am missing something, but can we use the pydoop hdfs apis instead of dataset_api, depending on hopsworks or local engine?

Reply:
I changed it to use hopsworks/local engine for copy and move. But I will keep the remove operation to go through the REST API so we can track it in the logs.